server/accounts.go

```diff
@@ -831,13 +872,34 @@ func (s *Server) updateAccountClaims(a *Account, ac *jwt.AccountClaims) {
 	s.Debugf("Updating account claims: %s", a.Name)
 	a.checkExpiration(ac.Claims())
 
+	a.mu.Lock()
 	// Clone to update, only select certain fields.
-	old := &Account{Name: a.Name, imports: a.imports, exports: a.exports, limits: a.limits}
+	old := &Account{Name: a.Name, imports: a.imports, exports: a.exports, limits: a.limits, signingKeys: a.signingKeys}
 
 	// Reset exports and imports here.
 	a.exports = exportMap{}
 	a.imports = importMap{}
 
+	// update account signing keys
+	a.signingKeys = nil
+	signersChanged := false
+	if len(ac.SigningKeys) > 0 {
+		// insure sorted and copy
+		sort.Strings(ac.SigningKeys)
+		a.signingKeys = append(a.signingKeys, ac.SigningKeys...)
```

Comment:
Ok, I am not sure if append(.., keys...) is optimized to the point of doing a single alloc of the known size. So you could do preallocate the array:
```
a.signingKeys = make([]string, len(ac.SigningKeys)
for i:=0; i<len(ac.SigningKeys); i++{
  a.signingKeys[i]=ac.SigningKeys[i]
}
```
or
```
a.signingKeys = make([]string, 0, len(ac.SigningKeys)
a.signingKeys = append(a.signingKeys, ac.SigningKeys...)
```
(I think I bench'ed this once, but forgot which one is faster ;-))

Also, should you sort the a.signingKeys instead of ac.SigningKeys?